lib/ansible/modules/windows/win_dns_client.ps1

```diff
@@ -153,36 +173,32 @@ Function Set-DnsClientAddresses
 {
     Param(
         [string] $adapter_name,
-        [string[]] $ipv4_addresses
+        [string[]] $dns_servers
     )
+    $dns_servers = [System.Net.IPAddress[]]$dns_servers
 
-    Write-DebugLog ("Setting DNS addresses for adapter {0} to ({1})" -f $adapter_name, ($ipv4_addresses -join ", "))
-    
-    If ($null -eq $ipv4_addresses) {
-        Set-DnsClientServerAddress -InterfaceAlias $adapter_name -ResetServerAddress
-    }
+    Write-DebugLog ("Setting DNS addresses for adapter {0} to ({1})" -f $adapter_name, ($dns_servers.IPAddressToString -join ", "))
 
-    Else {
-    # this silently ignores invalid IPs, so we validate parseability ourselves up front...
-        Set-DnsClientServerAddress -InterfaceAlias $adapter_name -ServerAddresses $ipv4_addresses
+    If ([bool]$dns_servers) {
+        Set-DnsClientServerAddress -InterfaceAlias $adapter_name -ResetServerAddress
+    } Else {
+        Set-DnsClientServerAddress -InterfaceAlias $adapter_name -ServerAddresses $dns_servers
     }
-    
-    # TODO: implement IPv6
 }
 
 $result = @{changed=$false}
 
 $params = Parse-Args -arguments $args -supports_check_mode $true
 
 $adapter_names = Get-AnsibleParam $params "adapter_names" -Default "*"
-$ipv4_addresses = Get-AnsibleParam $params "ipv4_addresses" -FailIfEmpty $result
+$dns_servers = Get-AnsibleParam $params "dns_servers" -FailIfEmpty $result
```

Comment:
I would keep both options for backward compatibility and document both as aliases

Reply:
I thought they are, isn't the alias stuff handled within the .py file already?

Reply:
No, the .py is just for documentation in windows, `Get-AnsibleParam` has an `-aliases` parameter